src/main/java/com/epam/ta/reportportal/core/item/impl/StartTestItemHandlerImpl.java

```diff
@@ -82,19 +84,15 @@ public ItemCreatedRS startRootItem(ReportPortalUser user, ReportPortalUser.Proje
 				.orElseThrow(() -> new ReportPortalException(LAUNCH_NOT_FOUND, rq.getLaunchId()));
 		validate(user, projectDetails, rq, launch);
 
-		TestItem item = new TestItemBuilder().addStartItemRequest(rq).addAttributes(rq.getAttributes()).addLaunch(launch).get();
-		Optional<TestItem> itemOptional;
-
-		//pzdc. need refactoring
-		if (launch.isRerun() && (itemOptional = testItemRepository.findByNameAndLaunchWithoutParents(rq.getName(),
-				launch.getId()
-		)).isPresent()) {
-			item = handleRerun(rq, launch, itemOptional.get(), null);
-		} else {
-			testItemRepository.save(item);
-			generateUniqueId(launch, item, String.valueOf(item.getItemId()));
+		Optional<ItemCreatedRS> rerun = rerunHandler.handleTestItem(rq, launch, null);
```

Comment:
it is better to give it a better name
rerunCreatedRs or something

Reply:
Also, maybe it is a good idea to call it only if rq has "rerun flag"

Reply:
Also, maybe it is good to create separate methods for handleRootItem and children
cause from the context it is hard to recognize what does this 'null' mean here

Reply:
fixed